src/main/scala/dotty/dokka/tasty/TastyParser.scala

```diff
@@ -38,7 +39,8 @@ case class TastyParser(reflect: Reflection, inspector: DokkaTastyInspector)
               docs += parsePackage(pck)
             case packageObject: ClassDef if(packageObject.symbol.name == "package$") =>
               docs += parsePackageObject(packageObject)
-            case clazz: ClassDef  =>
+            case clazz: ClassDef if(clazz.symbol.companionClass.exists && clazz.symbol.flags.is(Flags.Object)) =>
```

Comment:
We can move that if to rhs of case below:
```
 case clazz: ClassDef  =>
   if clazz.symbol.companionClass.exists && clazz.symbol.flags.is(Flags.Object) then
     docs += parseClass(clazz)
case _ =>
```

Reply:
Reversed condition and moved down